src/Main.elm

```diff
@@ -1,10 +1,829 @@
+                        ]
+                        <| E.text upperLeftText
+
+        upperLeftText : String
+        upperLeftText =
+            case model.sortBy of
+                Data.Audience.Shared ->
+                    "Displaying Shared audiences from all folders"
+
+                _ ->
+                    "Click a folder to start browsing!"
+
+        upperRightInfoElement : E.Element Msg
+        upperRightInfoElement =
+            E.row
+                [ EBackground.color Palette.color4
+                , E.padding 5
+                , E.spacing 5
+                , EBorder.roundEach { topRight = 99, bottomRight = 99, bottomLeft = 20, topLeft = 20 }
+                , EFont.color Palette.gray0
+                ]
+                [ E.paragraph
+                    [ E.paddingXY 5 0
+                    ]
+                    [ E.text "Currently sorting by "
+                    , E.el
+                        [ EFont.bold ]
+                        <| E.text (Data.Audience.audienceTypeToString model.sortBy)
+                    ]
+                , wrapColorCircle Palette.color3 coolI -- we wrap it in a colored circle
+                ]
+
+        -- the little panel above the file browser
+        upperPanel : E.Element Msg
+        upperPanel =
+            E.row
+                [ E.height <| E.px 50
+                , E.paddingEach { top = 0, right = 7, bottom = 0, left = 20 }
+                , EBackground.color Palette.gray4
+                , E.width E.fill
+                ]
+                [ upperLeftBackButtonOrText
+                , E.el [ E.alignRight ] upperRightInfoElement
+                ]
+
+        browser : E.Element Msg
+        browser =
+            E.column
+                [ E.width E.fill
+                , E.height E.fill
+                , EBackground.color Palette.gray4
+                , E.spacing 1
+                , EBorder.rounded 7
+                , EBorder.width 1
+                , EBorder.color Palette.color5
+
+                -- we do this, otherwise the elements inside aren't going to respect
+                -- our rounded borders
+                , E.htmlAttribute <| Html.Attributes.style "overflow" "hidden"
+                ]
+                [ upperPanel
+                , E.column
+                    [ E.width E.fill
+                    , E.height E.fill
+                    , E.spacing 1
+                    ]
+                    -- folders first, just like "FIRST_STEP.md" says
+                    (filteredAudienceFolders ++ filteredAudiences)
+                ]
+    in
+    E.row
+        [ E.spacing 20
+        ]
+        [ E.el [ E.alignTop, E.width <| E.px 700 ] browser
+        , viewSortingPanel model.sortBy
+        ]
+
+
+viewFolder : Int -> FormattedAudienceFolder -> List (E.Element Msg) -> List (E.Element Msg)
+viewFolder id folder acc =
+    let
+        -- we view a "folder" as a row: the icon, and the name of the folder
+        item =
+            E.row
```

Comment:
It's nice, that elm-ui is generating `role` for `button` types, but in case of list of items it's not that accessibility friendly for users as it looks from Elm code base.

Reply:
It does seem that for a purpose like this, you could use [Element.Region](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Region#heading) to "mark" some nodes, so elm-ui will make them "h1", or whatever you specify, although I'll be honest and say I haven't used it.